### PR TITLE
KAZOO-4107 Numbers that are 're-discovered' are not updated in the number database

### PR DESCRIPTION
**Pull request by lazedo**
KAZOO-4105 The number manger crawler does not appear to be crawling
KAZOO-4106 number crawler operates on chunks of documents

### CODE REVIEW

File: core/whistle_number_manager-1.0.0/include/wh_number_manager.hrl [hunk 1]
```diff
@@ -73,7 +73,7 @@
                                        ,<<"failover">> ,<<"prepend">>
                                       ]).
 
--define(WNM_DB_PREFIX_L, "numbers/").
+-define(WNM_DB_PREFIX_L, "numbers/+").
```
Comment (k-anderson) [inline core/whistle_number_manager-1.0.0/include/wh_number_manager.hrl]: @lazedo 
Why this change?  

Also, there is no guarantee that this is true.  If the number_manager e164 converters are configured differently on a cluster it may not result in a + prefix.  We have seen this before where a system had purposely been setup to result in US numbers without a '+1' prefix....

File: core/whistle_number_manager-1.0.0/src/wh_number_manager.erl [hunk 2]
```diff
@@ -1007,7 +1007,10 @@ prepare_find_results([Number|Numbers], ModuleName, ModuleResults, Found, Opts) -
                       | Found]
              end,
     case catch wnm_number:get(Number) of
-        #number{state=State} ->
+        #number{state=State
+                ,module_data=JObj
+                ,module_name = ModuleName
```
Comment (k-anderson) [inline core/whistle_number_manager-1.0.0/src/wh_number_manager.erl]: @lazedo 
Correct me if I am wrong, but it looks like the discovery is only re-saved if the module_name does not match?  IE: if a search is preformed by the same carrier module then it can not resave the number if it already exists because it will match this case-clause (which is the old behavior).

If so lets talk about it :)

File: core/whistle_number_manager-1.0.0/src/wnm_number_crawler.erl [hunk 3]
```diff
@@ -36,6 +36,10 @@
         ,whapps_config:get_integer(?WNM_CONFIG_CAT, <<"deleted_expiry_d">>, 90)
        ).
 
+-define(NUMBERS_TO_CRAWL
+        ,whapps_config:get_integer(?WNM_CONFIG_CAT, <<"number_of_documents_to_crawl">>, 1000)
```
Comment (k-anderson) [inline core/whistle_number_manager-1.0.0/src/wnm_number_crawler.erl]: @lazedo 
whistle_couch has a property default_chunk_size which I believe we have been using elsewhere.  I would prefer this as I don't see much value in changing the chunk size for individual operations.... The intent is to get chunks proportional to the size of the db cluster, so its very convenient to change it in one place...

File: core/whistle_number_manager-1.0.0/include/wh_number_manager.hrl (hunk 1 above)
Comment (lazedo) [inline core/whistle_number_manager-1.0.0/include/wh_number_manager.hrl]: @k-anderson there are some Dbs numbers/inum, numbers/managed numbers/local that were being crawled without reason. i understand your point here but e164 is meant to be +...

File: core/whistle_number_manager-1.0.0/src/wh_number_manager.erl (hunk 2 above)
Comment (lazedo) [inline core/whistle_number_manager-1.0.0/src/wh_number_manager.erl]: @k-anderson  if it matches this clause it will not be save, ie, if its the same module and the same module_data (JObj) then we don't save it (no need to).

another clause was added to save it in case there is a difference

File: core/whistle_number_manager-1.0.0/src/wnm_number_crawler.erl (hunk 3 above)
Comment (lazedo) [inline core/whistle_number_manager-1.0.0/src/wnm_number_crawler.erl]: @k-anderson got it, will use that one.

File: core/whistle_number_manager-1.0.0/src/wh_number_manager.erl (hunk 2 above)
Comment (k-anderson) [inline core/whistle_number_manager-1.0.0/src/wh_number_manager.erl]: @lazedo Ah overlooked JObj, that makes sense.

File: core/whistle_number_manager-1.0.0/include/wh_number_manager.hrl (hunk 1 above)
Comment (k-anderson) [inline core/whistle_number_manager-1.0.0/include/wh_number_manager.hrl]: @lazedo true but...still makes me nervous... if it is just for an efficiency with managed numbers I would prefer to remove it.  I think that carrier module is only used on dev boxes...
